src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java

```diff
@@ -280,55 +280,84 @@ else if (tsDiff >= (-(1L << 30) & 0xFFFFFFFFL))
                 {
-                    if (aLastSeqNum == -1)
+                    if (snDiff == 2 || snDiff == (-2 & 0xFFFF))
                     {
-                        aLastSeqNum = (aMaxSeqNum + 1) & 0xFFFF;
-                        if (logger.isDebugEnabled())
+                        if (aLastSeqNum == -1)
                         {
-                            logger.debug("Guessed frame end=" + aLastSeqNum);
+                            a.setEnd((aMaxSeqNum + 1) & 0xFFFF);
                         }
-                        a.setEnd(aLastSeqNum);
-                    }
-                    else
-                    {
-                        bFirstSeqNum = (bMinSeqNum - 1) & 0xFFFF;
-                        if (logger.isDebugEnabled())
+                        else
                         {
-                            logger.debug("Guessed frame start=" + bFirstSeqNum);
+                            b.setStart((bMinSeqNum - 1) & 0xFFFF);
                         }
-                        b.setStart(bFirstSeqNum);
+
+                        guessed = true;
                     }
                 }
-                else if (snDiff < 2 || snDiff > (-3 & 0xFFFF))
+                else // (bFirstSeqNum == -1 && aLastSeqNum == -1)
                 {
-                    logger.warn("Frame corruption or packets that are out of " +
-                        "order detected.");
+                    if (snDiff == 3 || snDiff == (-3 & 0xFFFF))
+                    {
+                        a.setEnd((aMaxSeqNum + 1) & 0xFFFF);
+                        b.setStart((bMinSeqNum - 1) & 0xFFFF);
+                        guessed = true;
+                    }
                 }
             }
             else
             {
-                if (snDiff == 3)
+                if (bFirstSeqNum != -1 || aLastSeqNum != -1)
                 {
-                    bFirstSeqNum = (bMinSeqNum - 1) & 0xFFFF;
-                    aLastSeqNum = (aMaxSeqNum + 1) & 0xFFFF;
-                    if (logger.isDebugEnabled())
+                    if (snDiff == 1 || snDiff == (-1 & 0xFFFF))
                     {
-                        logger.debug(
-                            "Guessed frame start=" + bFirstSeqNum
-                                + ",end=" + aLastSeqNum);
+                        if (aLastSeqNum == -1)
+                        {
+                            a.setEnd(aMaxSeqNum & 0xFFFF);
+                        }
+                        else
+                        {
+                            b.setStart(bMinSeqNum & 0xFFFF);
+                        }
+
+                        guessed = true;
                     }
+                }
+                else // (bFirstSeqNum == -1 && aLastSeqNum == -1)
+                {
+                    if (snDiff == 2 || snDiff == (-2 & 0xFFFF))
+                    {
+                        a.setEnd(aMaxSeqNum & 0xFFFF);
+                        b.setStart(bMinSeqNum & 0xFFFF);
```

Comment:
I don't understand this. If diff==2 and (bFirstSeqNum == -1 && aLastSeqNum == -1), then there is 1 packet between aLastSeen and bFirstSeen. And we don't know whether this packet belongs to a or to b, or is a separate frame of it's own (since in this if branch there is no support for frame boundaries, which means that e.g. aLastSeen could be the end of a even if aLastSeqNum == 1).